src/FormatNumber.elm

```diff
@@ -37,115 +95,61 @@ type alias Locale =
     >>> formatFloat (Locale 1 "," ".") -0.01
     "0.0"
 
+    >>> formatFloat (Locale 2 "," ".") 0.01
+    "0.01"
+
+    >>> formatFloat (Locale 0 "," ".") 123.456
+    "123"
+
+    >>> formatFloat (Locale 0 "," ".") 1e9
+    "1,000,000,000"
+
+    >>> formatFloat (Locale 5 "," ".") 1.0
+    "1.00000"
 -}
 formatFloat : Locale -> Float -> String
 formatFloat locale num =
-    let
-        multiplier : Int
-        multiplier =
-            10 ^ locale.decimals
-
-        digits : Int
-        digits =
-            num
-                |> (*) (toFloat multiplier)
-                |> round
-    in
-        formattedNumber locale digits
+    (formatInt locale (truncate num))
+        ++ (separator locale)
+        ++ (nDigits locale.decimals num)
 
 
 {-| Format a integer number as a pretty string:
 
     >>> formatInt { decimals = 1, thousandSeparator = ",", decimalSeparator = "." } 0
-    "0.0"
+    "0"
 
-    >>> formatInt (Locale 1 "," ".") 1234567890
-    "1,234,567,890.0"
+    >>> formatInt (Locale 1 " " ".") 1234567890
+    "1 234 567 890"
+
+    >>> formatInt (Locale 10 "," ".") -123456
+    "-123,456"
 
 -}
 formatInt : Locale -> Int -> String
 formatInt locale num =
-    num
-        |> toFloat
-        |> formatFloat locale
+    case compare num 0 of
+        LT ->
+            formatInt locale (-num) |> String.cons '-'
+
+        EQ ->
```

Comment:
> The user explicitly gave an Int 

Nope. The user explicitly passed an `Int` to a function that format it as `String` using a `Locale` that has a `decimals` field.

If user just wanted to print an `Int` it would be way easier to just pass `Int` to something else: e.g. `toString myIntegerNumber`.